demos/classification_demo/main.cpp

```diff
@@ -0,0 +1,537 @@
+        std::vector<unsigned> indices(tblob.size() / batchSize);
+        std::iota(std::begin(indices), std::end(indices), 0);
+        std::partial_sort(std::begin(indices), std::begin(indices) + numTop, std::end(indices),
+                          [&batchData](unsigned l, unsigned r) {
+                              return batchData[l] > batchData[r];
+                          });
+
+        indices.resize(numTop);
+        output[i].assign(indices.begin(), indices.begin() + numTop);
+    }
+    
+    return output;
+}
+
+int main(int argc, char *argv[]) {
+    try {
+        if (!ParseAndCheckCommandLine(argc, argv)) {
+            return 0;
+        }
+
+        // -----------------------------------------Read input images-----------------------------------------
+        std::vector<std::string> imageNames;
+        std::vector<cv::Mat> inputImages;
+        parseInputFilesArguments(imageNames);
+        if (imageNames.empty()) throw std::runtime_error("No images provided");
+        std::sort(imageNames.begin(), imageNames.end());
+        for (size_t i = 0; i < imageNames.size(); i++) {
+            const std::string& name = imageNames[i];
+            const cv::Mat& tmpImage = cv::imread(name);
+            if (tmpImage.data == nullptr) {
+                std::cerr << "Could not read image " << name << '\n';
+                imageNames.erase(imageNames.begin() + i);
+                i--;
+            } else {
+                inputImages.push_back(tmpImage);
+                size_t lastSlashIdx = name.rfind('/');
+                if (lastSlashIdx == std::string::npos) {
+                    lastSlashIdx = name.rfind('\\'); // for compatibility with Windows
+                }
+                imageNames[i] = name.substr(lastSlashIdx + 1);
+            }
+        }
+        // ---------------------------------------------------------------------------------------------------
+
+        // ----------------------------------------Read image classes-----------------------------------------
+        std::vector<unsigned> classIndices;
+        if (!FLAGS_gt.empty()) {
+            std::map<std::string, std::string> classIndicesMap;
+            std::ifstream inputGtFile(FLAGS_gt);
+            if (!inputGtFile.is_open()) throw std::runtime_error("Can't open the ground truth file.");
+            
+            std::string line;
+            while (std::getline(inputGtFile, line))
+            {
+                size_t separatorIdx = line.find(' ');
+                if (separatorIdx == std::string::npos) {
+                    throw std::runtime_error("The ground truth file has incorrect format.");
+                }
+                std::string imagePath = line.substr(0, separatorIdx);
+                size_t imagePathEndIdx = imagePath.rfind('/');
+                std::string classIndex = line.substr(separatorIdx + 1);
+                if (!std::all_of(classIndex.begin(), classIndex.end() - 1, ::isdigit)
```

Comment:
This check isn't really necessary. You already use `stoul` later to convert the index to a number. Just do it _here_ and it will throw if the number is invalid.